File: app/src/main/java/io/github/pshegger/infobook/adapters/pager/BLTCPagerAdapter.java
package io.github.pshegger.infobook.adapters.pager;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import io.github.pshegger.infobook.fragments.ExampleFragment;
import io.github.pshegger.infobook.fragments.NamedFragment;

/**
 * Created by pshegger on 2015.10.04..
 */
public class BLTCPagerAdapter extends FragmentPagerAdapter {
    private NamedFragment[] mFragments;

    public BLTCPagerAdapter(FragmentManager fm) {
        super(fm);

        mFragments = new NamedFragment[]{
                ExampleFragment.newInstance("BLTC"),
                ExampleFragment.newInstance("BLTC"),
                ExampleFragment.newInstance("BLTC")
        };
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return mFragments[position].getName();
    }

    @Override
    public Fragment getItem(int position) {
        return mFragments[position];
    }

    @Override
    public int getCount() {
        return mFragments.length;
    }
}

File: app/src/main/kotlin/io/github/pshegger/infobook/adapters/pager/BLTCPagerAdapter.kt
package io.github.pshegger.infobook.adapters.pager

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import io.github.pshegger.infobook.fragments.ExampleFragment

/**
 * Created by pshegger on 2015. 11. 10..
 */
class BLTCPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {
    private val fragments = arrayOf(
            ExampleFragment.newInstance("BLTC"),
            ExampleFragment.newInstance("BLTC"),
            ExampleFragment.newInstance("BLTC")
    )

    override fun getItem(position: Int): Fragment? = fragments[position]

    override fun getCount(): Int = fragments.size

    override fun getPageTitle(position: Int): CharSequence? = fragments[position].name()
}